prov/rxm/src/rxm_ep.c

```diff
@@ -404,6 +405,20 @@ static int rxm_ep_txrx_res_open(struct rxm_ep *rxm_ep,
 	if (ret)
 		goto err;
 
+	if (!fi_param_get_size_t(&rxm_prov, "sar_limit", &param)) {
+		if (param < rxm_info.tx_attr->inject_size) {
+			FI_WARN(&rxm_prov, FI_LOG_CORE,
+				"Requested SAR limit (%zd) is too small. "
+				"Setting it to %zd\n",
+				param, rxm_info.tx_attr->inject_size);
+			rxm_ep->sar_limit = rxm_info.tx_attr->inject_size;
```

Comment:
I think this effectively disables using SAR.  I'm not sure it matters if sar_limit ends up equal to inject_size or anything less than it in these cases.

Reply:
Yes, you're right this code disables using SAR protocol.
So, we can avoid resetting the provided value, but we should throw a warning that SAR protocol won't be used.